src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php

```diff
@@ -0,0 +1,709 @@
+            ;
+            $totalPrice = Tools::ps_round($unitPrice, 2) * ($product['product_quantity'] - $product['customizationQuantityTotal']);
+
+            $unitPriceFormatted = Tools::displayPrice($unitPrice, $currency);
+            $totalPriceFormatted = Tools::displayPrice($totalPrice, $currency);
+
+            $productsForViewing[] = new OrderProductForViewing(
+                $product['product_id'],
+                $product['product_name'],
+                $product['reference'],
+                $product['supplier_reference'],
+                $product['product_quantity'],
+                $unitPriceFormatted,
+                $totalPriceFormatted,
+                $product['current_stock'],
+                $this->imageTagSourceParser->parse($product['image_tag'])
+            );
+        }
+
+        return new OrderProductsForViewing($productsForViewing);
+    }
+
+    /**
+     * @param $pack_item
+     */
+    private function setProductImageInformation(&$pack_item): void
+    {
+        if (isset($pack_item['id_product_attribute']) && $pack_item['id_product_attribute']) {
+            $id_image = Db::getInstance()->getValue('
+                SELECT `image_shop`.id_image
+                FROM `' . _DB_PREFIX_ . 'product_attribute_image` pai' .
+                Shop::addSqlAssociation('image', 'pai', true) . '
+                WHERE id_product_attribute = ' . (int) $pack_item['id_product_attribute']);
+        }
+
+        if (!isset($id_image) || !$id_image) {
+            $id_image = Db::getInstance()->getValue(
+                '
+                SELECT `image_shop`.id_image
+                FROM `' . _DB_PREFIX_ . 'image` i' .
+                Shop::addSqlAssociation('image', 'i', true, 'image_shop.cover=1') . '
+                WHERE i.id_product = ' . (int) $pack_item['id_product']
+            );
+        }
+
+        $pack_item['image'] = null;
+        $pack_item['image_size'] = null;
+
+        if ($id_image) {
+            $pack_item['image'] = new Image($id_image);
+        }
+    }
+
+    /**
+     * @param Order $order
+     *
+     * @return OrderHistoryForViewing
+     */
+    private function getOrderHistory(Order $order): OrderHistoryForViewing
+    {
+        $history = $order->getHistory($this->context->language->id);
+
+        $statuses = [];
+
+        foreach ($history as $item) {
+            $statuses[] = new OrderStatusForViewing(
+                (int) $item['id_order_state'],
+                $item['ostate_name'],
+                $item['color'],
+                new DateTimeImmutable($item['date_add']),
+                (bool) $item['send_email'],
+                $item['employee_firstname'],
+                $item['employee_lastname']
+            );
+        }
+
+        return new OrderHistoryForViewing(
+            $order->current_state,
+            $statuses
+        );
+    }
+
+    private function getOrderDocuments(Order $order): OrderDocumentsForViewing
```

Comment:
We can refactor this into a dedicated class in a following PR

Reply:
Into separate Query/Handler you mean? :thinking:

Reply:
Not necessary. I think this function by itself is a relevant need. I can imagine a `OrderDocumentsManager` service in charge of manageing and loading order documents. The GetOrderForViewingHandler can use it (as an injected dependency) but another Handler or Core service could depend on it too